src/C++/Session.cpp

```diff
@@ -30,6 +30,28 @@
 
 namespace FIX
 {
+
+namespace
+{
+
+const std::map<int, std::string> kBusinessRejectToBusinessRejectText = {
+    std::pair<int, std::string>{BusinessRejectReason_OTHER, BusinessRejectReason_OTHER_TEXT},
+    std::pair<int, std::string>{BusinessRejectReason_UNKNOWN_ID, BusinessRejectReason_UNKNOWN_ID_TEXT},
+    std::pair<int, std::string>{BusinessRejectReason_UNKNOWN_SECURITY, BusinessRejectReason_UNKNOWN_SECURITY_TEXT},
+    std::pair<int, std::string>{BusinessRejectReason_UNKNOWN_MESSAGE_TYPE, BusinessRejectReason_UNSUPPORTED_MESSAGE_TYPE_TEXT},
+    std::pair<int, std::string>{BusinessRejectReason_APPLICATION_NOT_AVAILABLE, BusinessRejectReason_APPLICATION_NOT_AVAILABLE_TEXT},
+    std::pair<int, std::string>{BusinessRejectReason_CONDITIONALLY_REQUIRED_FIELD_MISSING, BusinessRejectReason_CONDITIONALLY_REQUIRED_FIELD_MISSING_TEXT},
+    std::pair<int, std::string>{BusinessRejectReason_NOT_AUTHORIZED, BusinessRejectReason_NOT_AUTHORIZED_TEXT},
+    std::pair<int, std::string>{BusinessRejectReason_DELIVERTO_FIRM_NOT_AVAILABLE_AT_THIS_TIME, BusinessRejectReason_DELIVERTO_FIRM_NOT_AVAILABLE_AT_THIS_TIME_TEXT},
+};
+
+std::string GetBusinessRejectText(const int& err) {
+  auto itr = kBusinessRejectToBusinessRejectText.find(err);
+  return itr != kBusinessRejectToBusinessRejectText.end() ? itr->second : "";
+}
+
+}
+
```

Comment:
Many of the BusinessReject messages currently are not hit with the current code.  Changed switch statement to a map implementation to cleanup unreachable code but to allow these cases to still be covered should someone add them in the future.